src/logging/logging.cpp

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <filesystem>
+#include <systemd/sd-journal.h>
+#undef LOG_ERR
+
+#include <logger/logmodule.hpp>
+#include <utils/exception.hpp>
+
+#include "logging.hpp"
+
+namespace aos::sm::logging {
+
+/***********************************************************************************************************************
+ * Utils
+ **********************************************************************************************************************/
+
+#define AOS_ERROR_WRAP_EX(err, msg) aos::Error(err, msg, __FILENAME__, __LINE__)
+
+/***********************************************************************************************************************
+ * Logging implementation.
+ **********************************************************************************************************************/
+
+void Logging::Init(
+    const config::LoggingConfig& config, InstanceIDProvider& instanceProvider, ILogReceiver& ILogReceiver)
+{
+    LOG_INF() << "Initializing new logging instance";
+
+    mConfig           = config;
+    mInstanceProvider = &instanceProvider;
+    mLogReceiver      = &ILogReceiver;
+    mClose            = false;
+
+    mWorkerThread = std::thread(&Logging::ProcessLogs, this);
```

Comment:
consider moving thread start into a separate Start public method.
So we can first initialize all needed objects  and only then start their logic.

At least it was useful for Zephyr

Reply:
lets keep a simpler interface for now, and extend if required.